edx_solutions_organizations/views.py

```diff
@@ -68,48 +74,52 @@ def list(self, request, *args, **kwargs):
-
-            # list of users with corresponding organizations to exclude
-            exclude_admin_users = [
-                Q(id=org_id) & Q(users__courseenrollment__user_id__in=users) for
-                org_id, users in admin_users_dict.items()]
-            if exclude_admin_users:
-                exclude_admin_users = reduce(lambda a, b: a | b, exclude_admin_users)
-                q_object.add(~Q(exclude_admin_users), Q.AND)
-
-            # filter only users courses that are enrolled i.e is_active=True.
-            q_object.add(Q(users__courseenrollment__is_active=True), Q.AND)
-
-            # annotating queryset to get number of courses
-            queryset = queryset.annotate(
-                number_of_courses=Count(
-                    Case(
-                        When(q_object, then=F('users__courseenrollment__course_id'))
-                    ), distinct=True
+        if str2bool(include_course_counts):
+            if exclude_type:
+                q_object = Q()
+
+                # Filtering roles to exclude
+                roles_to_exclude = [CourseInstructorRole.ROLE, CourseStaffRole.ROLE, CourseObserverRole.ROLE, CourseAssistantRole.ROLE]
+                user_ids = CourseAccessRole.objects.filter(role__in=roles_to_exclude).distinct().values_list('user_id', flat=True)
+                q_object.add(~Q(users__courseenrollment__user_id__in=user_ids), Q.AND)
+
+                # Filtering company admin users to exclude
+                admin_users = User.objects.filter(id__in=list(queryset.filter(
+                    Q(users__groups__groupprofile__name=exclude_type)
+                ).distinct().values_list('users', flat=True)))
+
+                admin_users_dict = {}
+                for user in admin_users:
+                    other_companies = list(user.organizations.all())[1:]
+                    for company in other_companies:
+                        admin_users_dict.setdefault(company.id, []).append(user.id)
+
+                # list of users with corresponding organizations to exclude
+                exclude_admin_users = [
+                    Q(id=org_id) & Q(users__courseenrollment__user_id__in=users) for
+                    org_id, users in admin_users_dict.items()]
+                if exclude_admin_users:
+                    exclude_admin_users = reduce(lambda a, b: a | b, exclude_admin_users)
+                    q_object.add(~Q(exclude_admin_users), Q.AND)
+
+                # filter only users courses that are enrolled i.e is_active=True.
+                q_object.add(Q(users__courseenrollment__is_active=True), Q.AND)
+
+                # annotating queryset to get number of courses
+                queryset = queryset.annotate(
+                    number_of_courses=Count(
+                        Case(
+                            When(q_object, then=F('users__courseenrollment__course_id'))
+                        ), distinct=True
+                    )
+                )
```

Comment:
This code is using from one endpoint of Apros where you have set this `include_course_counts `. It means this code should be removed if it is not using from Apros side.

Reply:
keeping this code for now to keep the functionality intact if it is being called somewhere else, in phase 2 which will come next month we will optimise this API.